plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveDynamicPartitionPruning.java

```diff
@@ -16,46 +16,31 @@
 import com.google.common.collect.ImmutableMap;
 import io.airlift.log.Logger;
 import io.trino.Session;
-import io.trino.spi.QueryId;
-import io.trino.spi.predicate.Domain;
-import io.trino.spi.predicate.ValueSet;
-import io.trino.testing.AbstractTestQueryFramework;
-import io.trino.testing.MaterializedResult;
+import io.trino.testing.AbstractDynamicFilteringIntegrationSmokeTest;
 import io.trino.testing.QueryRunner;
-import io.trino.testing.ResultWithQueryId;
 import org.intellij.lang.annotations.Language;
 import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
 
-import java.util.List;
-
-import static com.google.common.collect.Iterables.getOnlyElement;
-import static io.airlift.testing.Assertions.assertGreaterThan;
-import static io.airlift.testing.Assertions.assertGreaterThanOrEqual;
 import static io.airlift.units.Duration.nanosSince;
 import static io.trino.SystemSessionProperties.ENABLE_LARGE_DYNAMIC_FILTERS;
 import static io.trino.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
 import static io.trino.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
-import static io.trino.server.DynamicFilterService.DynamicFilterDomainStats;
-import static io.trino.server.DynamicFilterService.DynamicFiltersStats;
-import static io.trino.spi.predicate.Domain.none;
-import static io.trino.spi.predicate.Domain.singleValue;
-import static io.trino.spi.predicate.Range.range;
-import static io.trino.spi.type.BigintType.BIGINT;
-import static io.trino.spi.type.IntegerType.INTEGER;
 import static io.trino.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
 import static io.trino.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
 import static io.trino.tpch.TpchTable.getTables;
 import static java.lang.String.format;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertTrue;
 
 public class TestHiveDynamicPartitionPruning
-        extends AbstractTestQueryFramework
+        extends AbstractDynamicFilteringIntegrationSmokeTest
 {
     private static final Logger log = Logger.get(TestHiveDynamicPartitionPruning.class);
     private static final String PARTITIONED_LINEITEM = "partitioned_lineitem";
-    private static final long LINEITEM_COUNT = 60175;
+
+    @Override
+    protected boolean supportsNodeLocalDynamicFiltering()
+    {
+        return false;
```

Comment:
hive does support node local dynamic filtering

Reply:
@sopel39 I realized now that the fact that there can be connectors like hive which supports both node local and coordinator dynamic filtering makes my current change problematic. As we discussed in on another comment,  if I separate `AbstractDynamicFilteringIntegrationSmokeTest ` to node-local and coordinator bases, I would have to make tests for Hive like `TestHiveDynamicPartitionPruning` to derive from both bases, which is not possible in Java. So I will revert back to old state of having single Base class with methods like `supportsNodeLocalDynamicFiltering` and `supportsCoordinatorDynamicFiltering`. I will resubmit PR with that change. Kindly let me know if you see a better option.

Reply:
As discussed in other comments, tests have been moved to separate base classes , one for node local and one for coordinator based dynamic filter support. Hive will have two test classes testing each case and Memory will have one for node-local alone.